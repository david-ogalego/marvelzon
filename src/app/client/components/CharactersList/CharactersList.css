.charactersList {
    margin-top: 145px;
    padding-right: 5px;
    padding-left: 5px;
}

@media only screen and (min-device-width : 320px) and (max-device-width : 480px) {
    .charactersList {
        margin-top: 0px;
    }
}