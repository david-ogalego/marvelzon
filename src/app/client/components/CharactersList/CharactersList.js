import React from 'react';

const CharactersList = function(props) {
    return (
        <div className="characterList">
            <h3>Characters</h3>
            <ul>
                {props.characters.map((character) => 
                    <li>
                        Name: {character.name}
                        Description: {character.description}
                    </li>
                )
            }
            </ul>
        </div>
    );
}

export default CharactersList;