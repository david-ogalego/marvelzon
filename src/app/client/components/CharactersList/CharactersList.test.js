import React from 'react';
import CharactersList from './CharactersList';
import renderer from 'react-test-renderer';

const charactersList = [
    {
        name: 'David'
    },
    {
        name: 'Pepe'
    }
];

test('Snapshot CharactersList rendered empty', () => {
    const treeCharactersListEmpty = renderer.create(<CharactersList characters={[]} />).toJSON();
    expect(treeCharactersListEmpty).toMatchSnapshot();
});

test('Snapshot CharactersList rendered with characters', () => {
    const treeCharactersList = renderer.create(<CharactersList characters={charactersList} />).toJSON();
    expect(treeCharactersList).toMatchSnapshot();
});