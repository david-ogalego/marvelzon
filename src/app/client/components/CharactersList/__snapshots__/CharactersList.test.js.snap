// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot CharactersList rendered empty 1`] = `
<div
  className={undefined}
>
  <h3>
    Characters
  </h3>
  <ul />
</div>
`;

exports[`Snapshot CharactersList rendered with characters 1`] = `
<div
  className={undefined}
>
  <h3>
    Characters
  </h3>
  <ul>
    <li>
      David
    </li>
    <li>
      Pepe
    </li>
  </ul>
</div>
`;
